src/pages/mesh/advanced/secrets.md

```diff
@@ -0,0 +1,351 @@
+      }
+    ]
+  }
+}
+```
+
+#### `secrets.yaml`
+
+```yaml
+AEM: 
+  TOKEN: "my-token"
+  AEMUSERNAME: "user-name"
+  adminname: "admin-name"
+
+API_KEY: $COMMERCE_API_KEY
+```
+
+#### JSON file
+
+```json
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "type": "object",
+  "required": [
+    "headerKeys",
+    "headerValues",
+    "headers"
+  ],
+  "properties": {
+    "headerKeys": {
+      "type": "array",
+      "items": {
+        "type": "string"
+      }
+    },
+    "headerValues": {
+      "type": "array",
+      "items": {
+        "type": "string"
+      }
+    },
+    "headers": {
+      "type": "object"
+    }
+  }
+}
+```
+
+Use the following GraphQL query to retrieve the headers:
+
+<CodeBlock slots="heading, code" repeat="2" languages="json, json" />
+
+#### GraphQL
+
+```graphql
+{
+  data {
+    headers
+  }
+}
+```
+
+#### Response
+
+```json
+{
+  "data": {
+    "data": {
+      "headers": {
+        "accept-encoding": "gzip, br",
+        "cf-connecting-ip": "0.0.0.0",
+        "cf-ipcountry": "US",
+        "cf-ray": "abc123abc123",
+        "cf-visitor": "{\"scheme\":\"https\"}",
+        "connection": "Keep-Alive",
+        "host": "header-reflection-service",
+        "secretaemeader": "abcabcdefdefxyzxyz",
+        "secretheader": "\\/root",
+        "x-forwarded-proto": "https",
+        "x-real-ip": "0.0.0.0"
+      }
+    }
+  },
+  "extensions": {}
+}
+```
+
+### Authorization
+
+The following example provides a simple authorization test. This mesh only returns a valid response, if the `TOKEN` in the `secrets.yaml` file is also passed as an authorization header in the request. If the token does not match, the mesh will return an unauthorized error.
+
+The **JavaScript file** tab contains a more readable version of the Javascript that appears in the mesh in the `files` object.
+
+<CodeBlock slots="heading, code" repeat="3" languages="json, yaml, javascript" />
+
+#### `mesh.json`
+
+```json
+{
+  "meshConfig": {
+    "sources": [
+      {
+        "name": "Commerce",
+        "handler": {
+          "graphql": {
+            "endpoint": "https://venia.magento.com/graphql"
+          }
+        }
+      }
+    ],
+    "plugins": [
+      {
+        "hooks": {
+          "beforeAll": {
+            "composer": "./hooks.js#isAuth",
+            "blocking": true
+          }
+        }
+      }
+    ],
+    "files": [
+      {
+        "path": "./hooks.js",
+        "content": "module.exports = {\r\n  isAuth: ({ context }) => {\r\n    const { headers, secrets } = context;\r\n\r\n    if (headers.authorization != secrets.TOKEN) {\r\n      return {\r\n        status: \"ERROR\",\r\n        message: \"Unauthorized\",\r\n      };\r\n    } else {\r\n      return {\r\n        status: \"SUCCESS\",\r\n        message: \"Authorized\",\r\n      };\r\n    }\r\n  },\r\n};\r\n"
+      }
+    ]
```

Comment:
```suggestion
```

Reply:
@revanth0212 i don't think we should remove the embedded file from the config. this will make the mesh non-functional. at the very least we should reference an external file, right?

Reply:
The idea would be for someone using this to have the mesh and the required files in the same folder so the `aio api-mesh` commands can link automatically. Hence I was suggesting to remove the files content from the `mesh` config.

In this case the `./hooks.js` file is being referenced in `mesh.json` under plugin hooks.

If this could potentially confuse them, we can maybe mention that these files should be placed in the same folder for them to work.